subsys/shell/modules/kernel_service.c

```diff
@@ -73,9 +74,19 @@ static void shell_tdata_dump(const struct k_thread *thread, void *user_data)
 		      thread,
 		      tname ? tname : "NA");
 	shell_fprintf((const struct shell *)user_data, SHELL_NORMAL,
```

Comment:
if  you are willing to do a bit of cleanup you could create local shell variable instead of casting it in each call and use `shell_print` which looks more compact.

Reply:
Looks nicer. Thanks, done.